sdks/python/apache_beam/ml/transforms/handlers.py

```diff
@@ -0,0 +1,406 @@
+
+
+# TODO: Add metrics namespace.
+class TFTProcessHandler(ProcessHandler[ProcessInputT, ProcessOutputT]):
+  def __init__(
+      self,
+      *,
+      input_types: Optional[Dict[str, type]] = None,
+      output_record_batches=False,
+      transforms: List[_TFTOperation] = None,
+      namespace: str = 'TFTProcessHandler',
+  ):
+    """
+    A handler class for processing data with TensorFlow Transform (TFT)
+    operations. This class is intended to be subclassed, with subclasses
+    implementing the `preprocessing_fn` method.
+
+    Args:
+      input_types: A dictionary of column names and types.
+      output_record_batches: Whether to output RecordBatches instead of
+        dictionaries.
+      transforms: A list of transforms to apply to the data. All the transforms
+        are applied in the order they are specified. The input of the
+        i-th transform is the output of the (i-1)-th transform. Multi-input
+        transforms are not supported yet.
+      namespace: A metrics namespace for the TFTProcessHandler.
+    """
+    super().__init__()
+    self._input_types = input_types
+    self.transforms = transforms if transforms else []
+    self._input_types = input_types
+    self._output_record_batches = output_record_batches
+    self._artifact_location = None
+    self._namespace = namespace
+
+  def get_raw_data_feature_spec(
+      self, input_types: Dict[str, type]) -> dataset_metadata.DatasetMetadata:
+    """
+    Return a DatasetMetadata object to be used with
+    tft_beam.AnalyzeAndTransformDataset.
+    Args:
+      input_types: A dictionary of column names and types.
+    Returns:
+      A DatasetMetadata object.
+    """
+    raw_data_feature_spec = {}
+    for key, value in input_types.items():
+      raw_data_feature_spec[key] = self._get_raw_data_feature_spec_per_column(
+          typ=value, col_name=key)
+    raw_data_metadata = dataset_metadata.DatasetMetadata(
+        schema_utils.schema_from_feature_spec(raw_data_feature_spec))
+    return raw_data_metadata
+
+  def _get_raw_data_feature_spec_per_column(self, typ: type, col_name: str):
+    """
+    Return a FeatureSpec object to be used with
+    tft_beam.AnalyzeAndTransformDataset
+    Args:
+      typ: A type of the column.
+      col_name: A name of the column.
+    Returns:
+      A FeatureSpec object.
+    """
+    # lets conver the builtin types to typing types for consistency.
+    typ = native_type_compatibility.convert_builtin_to_typing(typ)
+    containers_type = (List._name, Tuple._name)
+    is_container = hasattr(typ, '_name') and typ._name in containers_type
+
+    if is_container:
+      dtype = typing.get_args(typ)[0]
+      if len(typing.get_args(typ)) > 1 or typing.get_origin(dtype) == Union:
+        raise RuntimeError(
+            f"Incorrect type specifications in {typ} for column {col_name}. "
+            f"Please specify a single type.")
+      if dtype not in _default_type_to_tensor_type_map:
+        raise TypeError(
```

Comment:
Does this check need to be applied in the `else` case as well? If so, should it be pulled out of the if/else?